### PR TITLE
crypto: thread: remove ossl_crypto_thread_native_terminate

### PR DESCRIPTION
**Pull request by ckalina**
##### Checklist

- [x] tests are added or updated

Currently `ossl_crypto_thread_native_terminate` is not required (neither for Argon2), remove it.

### CODE REVIEW

**Comment by t8m** [comment]
This resolves the remaining hang (on windows builds). So imo this is urgent.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
ping @openssl/committers for second review. This is urgent.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged since urgent.

**Pull request closed without merging**
